src/types/tuple.rs

```diff
@@ -375,6 +376,51 @@ impl<'py> Iterator for BoundTupleIterator<'py> {
         let len = self.len();
         (len, Some(len))
     }
+
+    #[inline]
+    #[cfg(not(feature = "nightly"))]
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        let length = self.length.min(self.tuple.len());
+        let target_index = self.index + n;
+        if target_index < length {
+            let item = {
+                {
+                    unsafe {
+                        BorrowedTupleIterator::get_item(self.tuple.as_borrowed(), target_index)
+                            .to_owned()
+                    }
+                }
+            };
```

Comment:
```suggestion
            let item = unsafe {
                BorrowedTupleIterator::get_item(self.tuple.as_borrowed(), target_index).to_owned()
            };
```
This looks a bit convoluted

Reply:
Thanks & revised